shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1601,23 +1635,63 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
   return CGRectZero;
 }
 
-- (CGRect)caretRectForPosition:(UITextPosition*)position {
-  // TODO(cbracken) Implement.
-
-  // As of iOS 14.4, this call is used by iOS's
-  // _UIKeyboardTextSelectionController to determine the position
-  // of the floating cursor when the user force touches the space
-  // bar to initiate floating cursor.
-  //
-  // It is recommended to return a value that's roughly the
-  // center of kSpacePanBounds to make sure the floating cursor
-  // has ample space in all directions and does not hit kSpacePanBounds.
-  // See the comments in beginFloatingCursorAtPoint.
-  return CGRectZero;
+- (BOOL)isRTLAtPosition:(NSUInteger)position {
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    if (_selectionRects[i].position == position) {
+      return _selectionRects[i].writingDirection == NSWritingDirectionRightToLeft;
+    }
+  }
+  return NO;
 }
 
-- (CGRect)bounds {
-  return _isFloatingCursorActive ? kSpacePanBounds : super.bounds;
+- (CGRect)caretRectForPosition:(UITextPosition*)position {
+  NSInteger index = ((FlutterTextPosition*)position).index;
+  UITextStorageDirection affinity = ((FlutterTextPosition*)position).affinity;
```

Comment:
yeah it's much cleaner when affinity is part of `FlutterTextPosition`. thx for the update.